sorc/regrid_sfc.fd/grids_IO.F90

```diff
@@ -0,0 +1,681 @@
+ extra = npets / n_tiles
+ do tile = 1, n_tiles
+   decomptile(:,tile)=(/1,extra/)
+ enddo
+
+ ! mosaic file
+ write(rchar,'(i5)') res_atm
+ fname = trim(dir_fix)//"/C"//trim(adjustl(rchar))// "_mosaic.nc"
+
+! create the grid
+ fv3_grid = ESMF_GridCreateMosaic(filename=trim(fname), &
+                                  regDecompPTile=decomptile, &
+                                  staggerLocList=(/ESMF_STAGGERLOC_CENTER, ESMF_STAGGERLOC_CORNER, &
+                                                   ESMF_STAGGERLOC_EDGE1, ESMF_STAGGERLOC_EDGE2/), &
+                                  indexflag=ESMF_INDEX_GLOBAL, &
+                                  tileFilePath=trim(dir_fix), &
+                                  rc=ierr)
+ if(ESMF_logFoundError(rcToCheck=ierr,msg=ESMF_LOGERR_PASSTHRU,line=__LINE__,file=__FILE__)) &
+    call error_handler("IN GridCreateMosaic", ierr)
+
+ end subroutine create_grid_fv3
+
+!> subroutine to create grid object for gaussian grids
+!!  also sets distribution across procs
+!! @param[in] grid_setup        data structure with grid details 
+!! @param[in] npets             total number of pets
+!! @param[in] localpet          local pet
+!! @param[out] gauss_grid       output ESMF grid 
+
+ subroutine create_grid_gauss(grid_setup, npets, localpet, gauss_grid)
+
+ ! INTENT IN
+ type(grid_setup_type), intent(in) :: grid_setup
+ integer, intent(in)   :: npets, localpet
+
+ ! INTENT OUT
+ type(esmf_grid)                   :: gauss_grid
+
+ type(esmf_polekind_flag)          :: polekindflag(2)
+ type(esmf_staggerloc)             :: staggerloc(2)
+
+ type(esmf_field)                  :: lon_fields(2)
+ type(esmf_field)                  :: lat_fields(2)
+ real(esmf_kind_r8), pointer       :: lon_ptr_field(:,:), lon_ptr_coord(:,:)
+ real(esmf_kind_r8), pointer       :: lat_ptr_field(:,:), lat_ptr_coord(:,:)
+
+ integer :: ierr, i_dim, j_dim, v
+
+ polekindflag(1:2) = ESMF_POLEKIND_MONOPOLE
+
+
+ i_dim = grid_setup%ires
+ j_dim = grid_setup%jres
+
+ if (localpet == 0) print*," creating gauss grid for ", i_dim, j_dim
+ gauss_grid = ESMF_GridCreate1PeriDim(minIndex=(/1,1/), &
+                                    maxIndex=(/i_dim,j_dim/), &
+                                    polekindflag=polekindflag, &
+                                    periodicDim=1, &
+                                    poleDim=2,  &
+                                    coordSys=ESMF_COORDSYS_SPH_DEG, &
+                                    regDecomp=(/1,npets/),  &
+                                    indexflag=ESMF_INDEX_GLOBAL, rc=ierr)
+ if(ESMF_logFoundError(rcToCheck=ierr,msg=ESMF_LOGERR_PASSTHRU,line=__LINE__,file=__FILE__)) &
+    call error_handler("IN GridCreate1PeriDim", ierr)
+
+ ! read lat lon coordinates into fields
+ call lonlat_read_into_fields(localpet, i_dim, j_dim,  &
+                         trim(grid_setup%fname_coord), trim(grid_setup%dir_coord), &
+                         grid_setup, gauss_grid, lon_fields, lat_fields)
+
```

Comment:
In this loop, you specify 'center' for v=1 and 'corner' for v=2. This is leading to the mismatch. All 'corner' arrays should contain one extra row with dimension 768x385. That is why I am seeing a j dimension of 65 on PET 5 and 64 on all other PETS.

Reply:
Nice work @GeorgeGayno-NOAA ! I'll get it fixed.